dask/core.py

```diff
@@ -43,6 +43,12 @@ def istask(x):
     >>> istask(1)
     False
     """
+    from dask.task_spec import BaseTask, LiteralTask
+
+    if isinstance(x, BaseTask):
+        if isinstance(x, LiteralTask):
+            return False
+        return True
     return type(x) is tuple and x and callable(x[0])
```

Comment:
It feels like this nested if-clause points out an inconsistency in the API (naming). x is a task if it's if's not a literal task. It could be that the ideal solution here would be renaming `istask` to something like `is_runnable_task` which I think I don't want to do here, but I wanted to point out that this feels weird.

Reply:
Agreed!

FWIW, I like `GraphNode`, `DataLiteral` (or even just `Data`/`Datum`), and `Task`. Feel _very_ free not to rename things right now though.